BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -379,6 +410,48 @@ AddSmbiosType (
 }
 
 
+/**
+  This function is called to initialize the SmbiosStringsPtr.
+**/
+VOID
+EFIAPI
+InitSmbiosStringPtr (
+  VOID
+  )
+{
+
+  EFI_STATUS            Status;
+  UINT16                Length;
+  VOID                 *SmbiosStringsPtr;
+
+  //
+  // Allocate the memmory for SMBIOS String Ptr,
+  //
+
+  SmbiosStringsPtr = AllocateZeroPool ((PcdGet16(PcdSmbiosStringsCnt) * sizeof (SMBIOS_TYPE_STRINGS)) );
+  if (SmbiosStringsPtr == NULL) {
+    DEBUG ((DEBUG_INFO, "SmbiosStringsPtr Memory allocation failed"));
+    ASSERT_EFI_ERROR(EFI_OUT_OF_RESOURCES);
+  }
+
+  //
+  // Copy contents from Default String Array
+  //
+
+  Length = sizeof (mDefaultSmbiosStrings);
+  if(Length <= (PcdGet16(PcdSmbiosStringsCnt) * sizeof (SMBIOS_TYPE_STRINGS)))

```

Comment:
I think the meaning of PcdSmbiosStringsCnt was changed to total items number available from actual items number. Please check other functions if need change accordingly.

Reply:
Already checked. it will not affect.

Reply:
As discussed, add changes in GetSmbiosString(), to check for End of SMBIOS Table and avoid the loop to check for all 64 items.

Reply:
Changes are done and code pushed.